src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -311,7 +312,7 @@ private ExpressionSyntax Transform( ExpressionSyntax expr )
 
 			SimpleLambdaExpressionSyntax lambExpr => lambExpr
 				.WithModifiers( RemoveAsyncModifier( lambExpr.Modifiers ) )
-				.WithParameter( Transform( lambExpr.Parameter ) )
+				.WithParameter( Transform( lambExpr.Parameter, removeCancellationToken: false ) ?? lambExpr.Parameter )

```

Comment:
Can we add a test showing why/the result of passing `false` here? Or do we already have that?

Reply:
Especially given the null coalescing operator. When does that come in to play?

Reply:
Added the test. The null coalescing operator shouldn't actually ever come into use if removeCancellationToken: false  but the compiler can't tell that the function can't return null if it's false so I had to add it

Reply:
Perhaps a separate function that can't return null would be better then? Or perhaps the logic should live in `TransformParams`?

Reply:
Thanks! Moving the logic to TransformParams seems cleaner